flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/NodeBuilder.java

```diff
@@ -298,6 +300,21 @@ public PropertiesBuilder expression(ExpressionNode expressionNode) {
             return this;
         }
 
+        public PropertiesBuilder<T> collection(Node expressionNode) {
+            semanticModel.typeOf(expressionNode).ifPresent(propertyBuilder::type);
```

Comment:
We do not need it here. just build the property with `.type(Property.Type.EXPRESSION)`

Reply:
@KavinduZoysa let's remove this line for now